### ISSUE TITLE
Add Language: Imba

### ISSUE THREAD

**Issue by familyfriendlymikey** [github/linguist#5910]
## Language name
[Imba](https://github.com/imba/imba)

## URL of example repository
[Imba.io repository](https://github.com/imba/imba.io)

## URL of syntax highlighting grammar
[Imba's vscode extension](https://github.com/imba/vscode-imba)

## Most popular extensions
`.imba`

## Detected language
Plain text

## Popularity
- Imba has [5,600 stars](https://github.com/imba/imba/stargazers) on GitHub.
- Imba has [498 dependent repositories](https://github.com/imba/imba/network/dependents) on GitHub.
- There are [2,864 .imba files](https://github.com/search?&q=extension%3Aimba) on GitHub.
- The popular programming course website [Scrimba](https://scrimba.com/) is written in Imba.
- If more proof of popularity/usage is required, I will provide more examples.

## Pull Request Information
- Here's [my fork](https://github.com/familyfriendlymikey/linguist/commit/ce6fb293dcbe04b702f4f347dd7ba83143f66c35), grammar compilation failed citing several PCRE errors, despite syntax highlighting working in VSCode.
- Here's the source of the [provided sample](https://github.com/familyfriendlymikey/game_of_life.imba), MIT license.

## Next Steps
In CONTRIBUTING.md it says
> If problems are found, please report them to the grammar maintainer

I would very much appreciate some guidance on next steps. Thank you.

**Comment by MystPi** [comment]
And I was about to suggest Imba too. Cool!

**Comment by lildude** [comment]
> I would very much appreciate some guidance on next steps. Thank you.

Sure...

> Here's [my fork](https://github.com/familyfriendlymikey/linguist/commit/ce6fb293dcbe04b702f4f347dd7ba83143f66c35), grammar compilation failed citing several PCRE errors, despite syntax highlighting working in VSCode.

If you're seeing errors it's because you're using Oniguruma-specific expressions and/or are using very long expressions. This should be clear in the errors. These work on VSCode because it uses that regex engine which isn't particularly strict, however GitHub uses PCRE for performance reasons. In order to resolve those, you'll need to switch out the Oniguruma-specific expressions for PCRE-compatible expressions and ensure your expressions aren't too long and adhere to PCRE's limits.

**Comment by somebee** [comment]
@familyfriendlymikey Maybe you could try pointing at
```
path = vendor/grammars/language-imba
url = https://github.com/somebee/language-imba
```
Instead of
```
path = vendor/grammars/vscode-imba
url = https://github.com/imba/vscode-imba.git
```

And see if it passes then? Would love to see imba included at some point!

**Comment by familyfriendlymikey** [comment]
@lildude Thank you very much for explaining. I've fixed the expressions and submitted a PR: #6080
